lib/board.rb

```diff
@@ -0,0 +1,82 @@
+class GameBoard
+  attr_accessor :board
+  def display_board
+    "
+      #{board[0]}  |  #{board[1]}  |  #{board[2]}\n
+    =================
+      #{board[3]}  |  #{board[4]}  |  #{board[5]}\n
+    =================
+      #{board[6]}  |  #{board[7]}  |  #{board[8]}\n"
+  end
+
+  def assign_symbol(parameter, player_one, player_two)
+    if parameter == 'X'
+      second_symbol = 'Y'
+      "#{player_one} selected #{parameter} and #{player_two} now takes #{second_symbol}"
+    elsif parameter == 'Y'
+      second_symbol = 'X'
+      "#{player_one} selected #{parameter} and #{player_two} now takes #{second_symbol}"
+    end
+  end
+
+  def check_winner(player_symbol)
+    return true if column_check(player_symbol) == true
+
+    return true if diag_check(player_symbol) == true
+
+    all_equal_row(board[0..2], player_symbol) ||
+      all_equal_row(board[3..5], player_symbol) ||
+      all_equal_row(board[6..8], player_symbol)
+  end
+
+  def column_check(player_symbol)
+    all_equal(board[0], player_symbol) &&
+      all_equal(board[3], player_symbol) &&
+      all_equal(board[6], player_symbol) ||
+      all_equal(board[1], player_symbol) &&
+        all_equal(board[4], player_symbol) &&
+        all_equal(board[7], player_symbol) ||
+      all_equal(board[2], player_symbol) &&
+        all_equal(board[5], player_symbol) &&
+        all_equal(board[8], player_symbol)
+  end
+
+  def diag_check(player_symbol)
+    all_equal(board[0], player_symbol) &&
+      all_equal(board[4], player_symbol) &&
+      all_equal(board[8], player_symbol) ||
+      all_equal(board[0], player_symbol) &&
+        all_equal(board[4], player_symbol) &&
+        all_equal(board[8], player_symbol)
+  end
```

Comment:
Take a look at this chunk of code. You've repeated the verification.

Reply:
The diag_check method has been changed with the correct statements to work properly.